source/blog/2021-03-29-dbal-2.13.md

```diff
@@ -0,0 +1,141 @@
+---
+title: "New Release: Doctrine DBAL 2.13 with Deprecations and Forward Compatibility"
+authorName: Benjamin Eberlei
+authorEmail: kontakt@beberlei.de
+permalink: /2021/03/29/dbal-2.13.html
+---
+
+We have released the last minor version in the 2.x family of Doctrine DBAL with
+additional forward compatibilty to DBAL 3 around `Statement` and `Result` classes
+and with an integration in our new deprecations logging library.
+
+In addition this DBAL release re-enables PHP 7.1 and 7.2 compatibility to give
+as much flexibility as possible to everyone with forward compatibility.
+
+### Statement and Result Forward Compatibility
+
+DBAL 3.0 extracts all fetch-methods from the `Statement` class and moved them
+to a new `Result` class that is returned from `Statement::execute`. We have
+backported this API to 2.13 - so that you can support writing code for both
+DBAL 2 and 3 at the same time.
+
+Old code:
+
+```php
+$statement = $connection->prepare('SELECT * FROM tbl WHERE col = ?');
+$statement->bindParam(1, $value);
+$statement->execute();
+
+while ($row = $statement->fetch()) {
```

Comment:
I know it's just an example but we've had a few iteration-related bugs in the DBAL where an empty value fetched from a recordset would break iteration. It should be `while (($row = $statement->fetch()) !== false)` to not set a bad example.